alchemist/alchemist-incarnation-biochemistry/src/main/java/it/unibo/alchemist/model/implementations/actions/RandomPolarization.java

```diff
@@ -0,0 +1,76 @@
+package it.unibo.alchemist.model.implementations.actions;
+
+import org.apache.commons.math3.random.RandomGenerator;
+import org.apache.commons.math3.util.FastMath;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import it.unibo.alchemist.model.implementations.positions.Continuous2DEuclidean;
+import it.unibo.alchemist.model.interfaces.Action;
+import it.unibo.alchemist.model.interfaces.CellNode;
+import it.unibo.alchemist.model.interfaces.Context;
+import it.unibo.alchemist.model.interfaces.Node;
+import it.unibo.alchemist.model.interfaces.Position;
+import it.unibo.alchemist.model.interfaces.Reaction;
+
+/**
+ * 
+ */
+public class RandomPolarization extends AbstractAction<Double> {
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = -1475157288544618036L;
+    @SuppressFBWarnings(value = "SE_BAD_FIELD", justification = "All provided RandomGenerator implementations are actually Serializable")
+    private final RandomGenerator rand;
```

Comment:
Evaluate the possibility of building an abstract class that includes a random field and protected methods to access it. Call it e.g. AbstractRandomizableAction, then make all the classes using a random inherit from it.

Always avoid code replication!